Python/compile.c

```diff
@@ -8641,18 +8649,31 @@ assemble(struct compiler *c, int addNone)
     if (label_exception_targets(g->g_entryblock)) {
         goto error;
     }
+
+    /** Optimization **/
+    consts = consts_dict_keys_inorder(c->u->u_consts);
+    if (consts == NULL) {
+        goto error;
+    }
     if (optimize_cfg(g, consts, c->c_const_cache)) {
         goto error;
     }
-    if (trim_unused_consts(g->g_entryblock, consts)) {
+    if (add_checks_for_loads_of_unknown_variables(g->g_entryblock, c) < 0) {
         goto error;
     }
+
+    /** line numbers (TODO: move this to desugaring stage) */
     if (duplicate_exits_without_lineno(g) < 0) {
         goto error;
     }
     propagate_line_numbers(g->g_entryblock);
     guarantee_lineno_for_exits(g->g_entryblock, c->u->u_firstlineno);
 
+    /** Assembly **/
+    if (push_cold_blocks_to_end(g, code_flags) < 0) {
```

Comment:
Why is this assembly, not optimization? This isn't needed for correctness, is it?

Reply:
Agree, this is optimization.